### PR TITLE
eeprom: Add driver for Serial Presence Detect (SPD) 5 EEPROM device and sample

### PR DESCRIPTION
**Pull request by finikorg**
Add driver for SPD5 and simple sample reading and decoding basic features of SPD5.

### CODE REVIEW

File: drivers/eeprom/eeprom_jesd300.c [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+	LOG_DBG("offset 0x%tx len %zd", (ptrdiff_t)offset, len);
+
+	/**
+	 * Using simple byte read allows to cross page boundaries,
+	 * TODO: Check block read possibility
+	 */
+	for (size_t i = 0; i < len; i++) {
+		read_byte(dev, offset + i, p + i);
+	}
```
Comment (henrikbrixandersen) [inline drivers/eeprom/eeprom_jesd300.c:137]: This sounds painfully slow for larger reads. Future patches for adding block reading will be welcome :)

File: samples/drivers/eeprom/boot_counter/README.rst [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.. _samples_eeprom:
+.. _samples_eeprom_boot_counter:
 
 EEPROM Sample
```
Comment (henrikbrixandersen) [inline samples/drivers/eeprom/boot_counter/README.rst:3]: Please change the title as well.

File: samples/drivers/eeprom/spd/intel_adl_rvp.overlay
Comment (henrikbrixandersen) [inline samples/drivers/eeprom/spd/intel_adl_rvp.overlay:1]: Would it make sense to add these devicetree nodes directly to `boards/x86/intel_adl`?

File: samples/drivers/eeprom/spd/intel_rpl_s_crb.overlay [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+		status = "okay";
+	};
+};
+
+/ {
+	aliases {
+		eeprom-spd = &eeprom0;
+	};
+};
```
Comment (henrikbrixandersen) [inline samples/drivers/eeprom/spd/intel_rpl_s_crb.overlay:21]: Same?

File: samples/drivers/eeprom/spd/src/spd.h
Comment (henrikbrixandersen) [inline samples/drivers/eeprom/spd/src/spd.h:1]: Should these defines be in a public header instead to allow using them outside the sample? Same goes for the parsing routines.

File: samples/drivers/eeprom/spd/sample.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+sample:
+  description: Serial Presence Detect (SPD) EEPROM sample
+  name: spd sample
+common:
+  platform_allow: intel_rpl_s_crb intel_adl_rvp
+  build_only: true
```
Comment (henrikbrixandersen) [inline samples/drivers/eeprom/spd/sample.yaml:6]: I would prefer a build-only test added in `tests/drivers/build_all/eeprom/` and change this sample to allow for run-time verification (e.g. by matching the expected sample output using a console harness).

**Review by henrikbrixandersen — CHANGES_REQUESTED** [review]

**Review by kartben — CHANGES_REQUESTED** [review]
Please don't forget to add a README for the new sample

**Comment by finikorg** [comment]
> @finikorg Any plans resuming work on this? It would be nice to get this in v3.6.0.

I am still thinking how to make I2C-like block transfer working with SMBus, since SMBus block operations are not working with this EEPROM.

**Pull request closed without merging**